tests/DataSourceSimpleTests.js

```diff
@@ -24,11 +24,16 @@ require("./shared/DataSourceTestUtils.js");
 
 kettle.tests.dataSource.ensureDirectoryEmpty("%kettle/tests/data/writeable");
 
+/** KETTLE-34 options resolution and merging test **/
+
 fluid.defaults("kettle.tests.KETTLE34dataSource", {
     gradeNames: "kettle.dataSource.URL",
     url: "https://user:password@thing.available:997/path",
     headers: {
         "x-custom-header": "x-custom-value"
+    },
+    censorRequestOptionsLog: {
+        auth: false
```

Comment:
Please add a test for `auth: true` with sensitive values being replaced.

Reply:
But that is the default value - is the idea just to test that options merging works correctly?